src/shop/tests/test_api.py

```diff
@@ -391,3 +404,80 @@ def test_update_quantity_prediction(self, predict_quantity_mock):
         predict_quantity_mock.assert_called_once_with(product.id, target=0)
         product.refresh_from_db()
         self.assertEqual(product.out_of_stock_forecast, date(1337, 1, 1))
+
+    @mock.patch('shop.suppliers.get_supplier_api')
+    def test_order_from_supplier(self, mock_get_supplier_api):
+        mock_supplier_api = mock.MagicMock()
+        mock_get_supplier_api.return_value = mock_supplier_api
+        mock_order_product = mock_supplier_api.order_product
+        supplier1 = factories.SupplierFactory()
+        supplier2 = factories.SupplierFactory()
+        product1 = factories.ProductFactory()
+        product2 = factories.ProductFactory()
+        sp1 = factories.SupplierProductFactory(
+            product=product1,
+            supplier=supplier1,
+            price=5,
+            units=64
+        )
+        sp2 = factories.SupplierProductFactory(
+            product=product1,
+            supplier=supplier1,
+            price=10,
+            units=30
+        )
+        sp3 = factories.SupplierProductFactory(
+            product=product1,
+            supplier=supplier1,
+            price=40,
+            qty_multiplier=10,
+            units=1
+        )
+        factories.SupplierProductFactory(
+            product=product2,
+            supplier=supplier2,
+            price=1,
+            units=1
+        )
+        sp5 = api.order_from_supplier(product1.id, 48)
```

Comment:
In my eyes, there is some black magic going on here, this should raise an exception.